### PR TITLE
Update user name in put API

### PR DESCRIPTION
**Pull request by lvreynoso**
### Summary:
- **What:** Allows the updating of a user's name in an API endpoint. Adds in option in the auth0 sync script to update user attributes.
- **Ticket:** [sc202623](https://app.shortcut.com/genepi/story/202623)
- **Env:** [https://putuser-frontend.dev.czgenepi.org](https://putuser-frontend.dev.czgenepi.org)

### Checklist:
- [x] I merged latest `<base branch>`
- [x] I manually verified the change
- [x] I added labels to my PR
- [x] I tested in multiple browsers
- [x] I added relevant unit tests
- [x] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/backend/aspen/cli/sync_auth0.py [hunk 1]
```diff
@@ -28,16 +28,22 @@ def __init__(self, auth0_client: Auth0Client, db: Session, dry_run: bool) -> Non
         self.db = db
         self.dry_run = dry_run
 
-    def auth0_delete(self, ojbect: Any) -> None:
+    def auth0_delete(self, object: Any) -> None:
```
Comment (jgadling) [inline src/backend/aspen/cli/sync_auth0.py:34]: 😭 oops

File: src/backend/aspen/api/schemas/usergroup.py [hunk 2]
```diff
@@ -37,6 +37,7 @@ class UserBaseResponse(BaseResponse):
 class UserUpdateRequest(BaseRequest):
     agreed_to_tos: Optional[bool] = None
     acknowledged_policy_version: Optional[datetime.date] = None
+    name: Optional[str]
```
Comment (jgadling) [inline src/backend/aspen/api/schemas/usergroup.py]: does this need an `= None` ?

File: src/backend/aspen/api/views/users.py [hunk 3]
```diff
@@ -27,15 +28,24 @@ async def get_current_user(
...
     if user_update_request.acknowledged_policy_version is not None:
         user.acknowledged_policy_version = (
             user_update_request.acknowledged_policy_version
         )
+    if user_update_request.name is not None:
+        user.name = user_update_request.name
+        auth0_update_items["name"] = user_update_request.name
     await db.commit()
+
+    if user.auth0_user_id:
```
Comment (jgadling) [inline src/backend/aspen/api/views/users.py]: Should this check whether `auth0_update_items` is empty?

File: src/backend/aspen/api/schemas/usergroup.py (hunk 2 above)
Comment (lvreynoso) [reply src/backend/aspen/api/schemas/usergroup.py]: I don't think it *needs* it, but I can put one there,

**Review by jgadling — APPROVED** [review]

**Pull request merged**